apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -253,6 +257,258 @@ defmodule Astarte.AppEngine.API.Device do
+              "Tried to publish on invalid path #{inspect(path)} for object aggregated " <>
+                "interface #{inspect(interface_descriptor.name)}",
+              tag: "invalid_path"
+            )
+
+            {:error, :mapping_not_found}
+
+          {:error, :invalid_object_aggregation_path} ->
+            Logger.warn(
+              "Tried to publish on invalid path #{inspect(path)} for object aggregated " <>
+                "interface #{inspect(interface_descriptor.name)}",
+              tag: "invalid_path"
+            )
+
+            {:error, :mapping_not_found}
+        end
+    end
+  end
+
+  defp check_object_aggregation_prefix(path, guessed_endpoints, mappings) do
+    received_path_depth = path_or_endpoint_depth(path)
+
+    Enum.reduce_while(guessed_endpoints, :ok, fn
+      endpoint_id, _acc ->
+        with {:ok, %Mapping{endpoint: endpoint}} <- Map.fetch(mappings, endpoint_id),
+             endpoint_depth when received_path_depth == endpoint_depth - 1 <-
+               path_or_endpoint_depth(endpoint) do
+          {:cont, :ok}
+        else
+          _ ->
+            {:halt, {:error, :invalid_object_aggregation_path}}
+        end
+    end)
+  end
+
+  defp update_object_interface_values(
+         client,
+         realm_name,
+         device_id,
+         interface_descriptor,
+         path,
+         raw_value
+       ) do
+    {:ok, mappings} = Mappings.fetch_interface_mappings(client, interface_descriptor.interface_id)
+
+    timestamp_micro =
+      DateTime.utc_now()
+      |> DateTime.to_unix(:microsecond)
+
+    with {:ok, endpoint} <-
+           resolve_path(path, interface_descriptor, mappings),
+         endpoint_id <- endpoint.endpoint_id,
+         expected_types <-
+           extract_expected_types(path, interface_descriptor, mappings),
+         :ok <- validate_value_type(expected_types, raw_value) do
+      Queries.insert_value_into_db(
+        client,
+        device_id,
+        interface_descriptor,
+        nil,
+        nil,
+        path,
+        raw_value,
+        timestamp_micro
+      )
+
+      wrapped_value = wrap_to_bson_struct(nil, raw_value)
+
+      Queries.insert_path_into_db(
+        client,
+        device_id,
+        interface_descriptor,
+        endpoint_id,
+        path,
+        timestamp_micro,
+        div(timestamp_micro, 1000),
+        []
+      )
+
+      DataTransmitter.push_datastream(
+        realm_name,
+        device_id,
+        interface_descriptor.name,
+        path,
+        wrapped_value
+      )
+
+      {:ok,
+       %InterfaceValues{
+         data: raw_value
+       }}
+    else
+      {:error, :unexpected_value_type, expected: value_type} ->
+        Logger.warn("Unexpected value type.", tag: "unexpected_value_type")
+        {:error, :unexpected_value_type, expected: value_type}
+
+      {:error, :invalid_object_aggregation_path} ->
+        Logger.warn("Error while trying to pubblish on path for object aggregated",
```

Comment:
typo: publish.
Also "object aggregated interface", and try to always end with a period